Proposal_RepetitionTime.md

```diff
@@ -16,45 +16,55 @@ See Common MR metadata fields for a list of terms and their definitions.
 
 There are also some OPTIONAL JSON fields specific to anatomical scans:
 
-* `RepetitionTimeExcitation`: The time in seconds between successive excitation pulses that excite the same tissue.
-As with `RepetitionTimeInversion` this corresponds to [DICOM Tag 0018, 0080](http://dicomlookup.com/lookup.asp?sw=Tnumber&q=(0018,0080)): "the period of time … between the beginning of a pulse sequence and the beginning of the succeeding (essentially identical) pulse sequence".
-Note that although this would typically be called `RepetitionTime` please use `RepetitionTimeExcitation` for structural scans with multiple excitations as `RepetitionTime` is already defined as the amount of time that it takes to acquire a single volume in section 8.3.3 and to distinguish it from `RepetitionTimeInversion`.
+* `RepetitionTimeExcitation`: The time in seconds between successive excitation pulses that excite the same tissue. The DICOM tag best refers to this parameter is [(0018, 0080)](http://dicomlookup.com/lookup.asp?sw=Tnumber&q=(0018,0080)): "the period of time … between the beginning of a pulse sequence and the beginning of the succeeding (essentially identical) pulse sequence". This may be superseded by `RepetitionTimePreparation` for certain use cases, such as [MP2RAGE](https://infoscience.epfl.ch/record/172927/files/mp2rage.pdf). Please see the description of `RepetitionTimePreparation` parameter for further information. Finally, please note that `RepetitionTimeExcitation` (0018,0080) is also widely refered to as `RepetitionTime` in MRI physics literature. However, `RepetitionTime` is already defined as the amount of time that it takes to acquire a single volume in section 8.3.3. Therefore, please use `RepetitionTimeExcitation` (along with `RepetitionTimePreparation`, if needed) for anatomy imaging data.
```

Comment:
since you are defining this as a new key, would it make sense to keep the units the same as DICOM in `ms`?

Reply:
As other timing parameters in the main spec are described in seconds, I would say we keep it in seconds here for the sake of consistency.

Reply:
Yep :+1: on @agahkarakuzu's comment here. Here's the reference to the section in BIDS that says we have to convert to `s` from `ms`: https://github.com/bids-standard/bids-specification/blob/master/src/02-common-principles.md#units

Reply:
@KirstieJane, @agahkarakuzu  - it was a suggestions since this PR is already deviating from the spec with respect to things that needed to be added. 

do note the wording says `SHOULD` not `MUST` (even though it later says `needs to`), which means there is a weighing in place to be considered. 

For certain things when the scale is in micro or nano secs, i don't think it makes sense to convert to seconds. BIDS being a text-based standard vs a binary standard actually results in representing float as text, which has all kinds of issues. representing numbers as integers when possible has several benefits. thus far with the focus on MR, and more specifically fMRI, most units are in s. As this and other BEPs from other modalities get merged, the units of consideration need to be weighed.

The specific DICOM parameter the section is referring to is `RepetitionTime` and the decision at that time was taken with fMRI in mind (80% use case), where the typical natural order is in seconds. the original DICOM standard was not created with fMRI in mind. it was closer to the MR physics definition of excitation. Since this PR is an attempt to bring those pieces into play, it would be good to consider and perhaps note why choosing `s` over a natural scale was more appropriate.

Reply:
In any practical case I can think of, `RepetitionTimeExcitation` has to be a float value. If you specified it in milliseconds, integers would not be sufficient. I frequently have TRs of e.g. 8.5ms. I think most would agree that microseconds are not a natural scale for anyone other than sequence programmers (I certainly do not want to have to write TR=8500 microseconds and then sprinkle `1.e-6` conversion factors through my code). I have never needed nanosecond precision for MR. To my knowledge the finest time resolution on any existing scanner is 2 microseconds.

Hence in my opinion seconds are the natural scale for MR. Keeping the units in seconds massively reduces the number of conversion factors you have to add to equations, which is a common source of bugs (for me anyway).

Reply:
@spinicist - fair enough - the point of this discussion is to ensure that people do not choose a time unit just because, but because it is the appropriate thing to do so.  it is critical that once the choice is made we do not change it, but let's ensure the choice is made with a wide range of use cases in mind. this discussion should be had not just for this particular key, for any key that is being introduced in this bep, and in the spec more generally. the intent of the `SHOULD` is that this is recommended, but you (the "royal you" creators of a new field) should still decide if it is appropriate.

regarding conversions people have to either change existing code or convert from BIDS units one way or the other. so that is less of a concern. people should be careful in their code (just ask NASA about their rover).

Reply:
Hi @satra & @spinicist! So sorry for my slow reply.

I like @spinicist's point about not wanting to write `1.e-6` everywhere. I agree that the repetition times are likely to be most humanly readable in seconds.

I had implicitly interpreted the `SHOULD` slightly differently to @satra - I had read it as "unless you have a good reason not to (for example if we were including values in years!). This is different to the idea that the royal "you" 👑 needs to make an informed decision every time.

My position is that we _have_ had the conversation. We've decided to keep seconds. And I propose that we do not include the justification in the documentation anywhere.

So specifically, can I mark this conversation as resolved? Do we need to add anything to the specification?

Reply:
@KirstieJane - a lot of bids came out of the cognitive neuroimaging world where fMRI rules supreme. bep001 comes out of tissue quantification world, and in that world many of the sequences are more tuned to other things, and we will continue to see changes.  in this world, times are represented typically in different units:

https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0169265

i'm fine with marking this conversation as resolved, but i do want people to consider things beyond a few years or a few sequences. my point was to ensure that people think through before setting in stone, which is often what a standard is.  some things are much harder to change than others, and thus require more careful consideration.